### ISSUE TITLE
Code style guidelines

### ISSUE THREAD

**Issue by MuffinTastic** [xezno/mocha#35]
It's easy enough to follow the implicit style set by the existing C++ code, but we should probably explicitly set some convention guidelines for C++ just as we do for C#.

Also, do we have any guidelines in mind for what kinds of functionality should go approximately where on the managed side after bindings are made? In #34 I'd like to add something that actually calls the `Root::Quit()` binding, but I'm (as of writing this) honestly not entirely sure where it should go (Common or Engine, etc). It'll probably happen again, so it would be nice to have rough ideas for the future

**Comment by xezno** [comment]
Agreed, I'm going to start cleaning stuff up and documenting it. I'm probably going to set up something like [ReadTheDocs](https://readthedocs.org/) for this since it's battle tested and I find it better than github's wiki system, though I'm open to other suggestions!

**Comment by xezno** [comment]
I started some docs over on [the mocha-docs repo](https://github.com/xezno/mocha-docs) but I haven't had enough time today to add anything meaningful to them. It currently uses markdown but previewing stuff is a pain in the ass if you don't have ruby installed, so I'm going to look for something that offers a browser editor instead.

**Comment by MuffinTastic** [comment]
Looks good. Imo it's easy enough to use GitHub Codespaces (VSCode in your browser) to edit it, it has ruby and jekyll and bundler and everything else you might want pre-installed, no installation fuss involved. No complaints about something even easier though

**Comment by xezno** [comment]
Moved the wiki over to MediaWiki because it's easier to edit, should just be able to make an account on there and edit stuff: https://wiki.gu3.me/index.php/Main_Page. Will swap the domain out for something better at a later date
